purchases/index.js

```diff
@@ -505,8 +505,16 @@ function renderHtmlReceipt(purchase) {
     return mustache.render(receiptEmailLayout, purchase.data.viewModel);
 }
 
-function listPurchases(db, callback) {
-    db.query("SELECT * FROM purchase", (error, result) => {
+function listPurchases(db, options, callback) {
+    if(!callback){
+        callback = options
+        options = {};
+    }
+   
+    const offset = getOffset(options);
+    const limit = getLimit(options);
```

Comment:
Generally we want the *parsing* of data to happen at the endpoint level. This method is business logic, so we should be able to assume that parameters used to call methods here are *reasonable*.